### PR TITLE
getting-started-with-teswiz Assignment

### PR DESCRIPTION
**Pull request by ishant002**

### CODE REVIEW

File: src/test/java/com/znsio/sample/e2e/businessLayer/amazon/IphoneListPageBL.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+import com.context.TestExecutionContext;
+import com.znsio.e2e.entities.Platform;
+import com.znsio.e2e.runner.Runner;
+import com.znsio.sample.e2e.entities.SAMPLE_TEST_CONTEXT;
+import com.znsio.sample.e2e.screen.amazon.IphoneListScreen;
+import org.apache.log4j.Logger;
+
+import static org.assertj.core.api.Assertions.assertThat;
+public class IphoneListPageBL {
```
Comment (poojamanna18) [inline src/test/java/com/znsio/sample/e2e/businessLayer/amazon/IphoneListPageBL.java:11]: Name of the class is not relevant SearchResultListPageBL

File: src/test/java/com/znsio/sample/e2e/businessLayer/amazon/ProductPageBL.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+                .getId();
+        this.context = Runner.getTestExecutionContext(threadId);
+        this.currentUserPersona = SAMPLE_TEST_CONTEXT.ME;
+        this.currentPlatform = Runner.platform;
+    }
+
+    public ProductPageBL verifyProductDetails() {
+        String getProductName = ProductPageScreen.get().verifyProductName();
+        assertThat(getProductName).containsIgnoringCase("iphone 13");
+//        boolean isShareButtonPresent = ProductPageScreen.get().verifyIfShareButtonIsPresent();
```
Comment (poojamanna18) [inline src/test/java/com/znsio/sample/e2e/businessLayer/amazon/ProductPageBL.java:38]: remove commented line of codes

File: src/test/java/com/znsio/sample/e2e/entities/SAMPLE_TEST_CONTEXT.java [hunk 3]
```diff
@@ -8,4 +8,6 @@
     public static final String MEETING_ID = "meetingId";
     public static final String MEETING_PASSWORD = "meetingPassword";
     public static final String INVITATION_LINK = "invitationLink";
+    public static final String USER_NAME = "raghav.gargril291900@gmail.com";
```
Comment (poojamanna18) [inline src/test/java/com/znsio/sample/e2e/entities/SAMPLE_TEST_CONTEXT.java:11]: Username should be passed from BL or step class
Context class and its variable are used for the Test level data crawling

File: src/test/java/com/znsio/sample/e2e/screen/web/amazonWeb/AmazonLogInScreenWeb.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+import com.znsio.sample.e2e.screen.web.jiomeet.InAMeetingScreenWeb;
+import org.apache.log4j.Logger;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+
+public class AmazonLogInScreenWeb extends AmazonLogInScreen {
+    private final Driver driver;
+    private final Visual visually;
+    private final WebDriver innerDriver;
+    private static final String SCREEN_NAME = InAMeetingScreenWeb.class.getSimpleName();
```
Comment (poojamanna18) [inline src/test/java/com/znsio/sample/e2e/screen/web/amazonWeb/AmazonLogInScreenWeb.java:19]: name of the class is incorrect

File: src/test/java/com/znsio/sample/e2e/businessLayer/amazon/AmazonHomePageBL.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        long threadId = Thread.currentThread()
+                .getId();
+        this.context = Runner.getTestExecutionContext(threadId);
+        this.currentUserPersona = SAMPLE_TEST_CONTEXT.ME;
+        this.currentPlatform = Runner.platform;
+    }
+
+    public SearchResultListPageBL searchForIphone13() {
+        AmazonHomeScreen.get().searchIphone13();
+        return new SearchResultListPageBL();
```
Comment (vishal07111998) [inline src/test/java/com/znsio/sample/e2e/businessLayer/amazon/AmazonHomePageBL.java:35]: no need to create object of same class

File: src/test/java/com/znsio/sample/e2e/screen/amazon/ProductPageScreen.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+        switch(platform) {
+            case web:
+                return new ProductPageScreenWeb(driver, visually);
+        }
+        throw new NotImplementedException(SCREEN_NAME + " is not implemented in " + Runner.platform);
+    }
+
+    public abstract String verifyProductName();
+//    public abstract boolean verifyIfShareButtonIsPresent();
```
Comment (vishal07111998) [inline src/test/java/com/znsio/sample/e2e/screen/amazon/ProductPageScreen.java:35]: remove the commented code

File: src/test/resources/testData.json [hunk 7]
```diff
@@ -7,6 +7,10 @@
       "password": "eotTesting@0",
```
Comment (vishal07111998) [inline src/test/resources/testData.json:7]: remove the unused data

File: src/test/resources/environments.json [hunk 8]
```diff
@@ -4,6 +4,7 @@
     "BASE_URL": "http://the-internet.herokuapp.com/",
     "IMAGES_BASE_URL": "http://images.google.com/",
     "BING_BASE_URL": "http://bing.com/images",
-    "JIOMEET_BASE_URL": "https://jiomeetpro.jio.com/home"
+    "JIOMEET_BASE_URL": "https://jiomeetpro.jio.com/home",
+    "AMAZON_BASE_URL": "https://www.amazon.in/ap/signin?openid.pape.max_auth_age=0&openid.return_to=https%3A%2F%2Fwww.amazon.in%2F%3Fref_%3Dnav_custrec_signin&openid.identity=http%3A%2F%2Fspecs.openid.net%2Fauth%2F2.0%2Fidentifier_select&openid.assoc_handle=inflex&openid.mode=checkid_setup&openid.claimed_id=http%3A%2F%2Fspecs.openid.net%2Fauth%2F2.0%2Fidentifier_select&openid.ns=http%3A%2F%2Fspecs.openid.net%2Fauth%2F2.0&"
```
Comment (vishal07111998) [inline src/test/resources/environments.json:8]: use it the base amazon url not sign in page

File: src/test/java/com/znsio/sample/e2e/businessLayer/amazon/AmazonCartBL.java [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    public AmazonCartBL() {
+        long threadId = Thread.currentThread()
+                .getId();
+        this.context = Runner.getTestExecutionContext(threadId);
+        this.currentUserPersona = SAMPLE_TEST_CONTEXT.ME;
+        this.currentPlatform = Runner.platform;
+    }
+    public AmazonCartBL verifyAmIOnCart() {
```
[the remark this answers is not in the document]
Comment (ishant002) [reply src/test/java/com/znsio/sample/e2e/businessLayer/amazon/AmazonCartBL.java]: @vishal07111998 it's already in camel case

File: src/test/java/com/znsio/sample/e2e/businessLayer/amazon/AmazonHomePageBL.java (hunk 5 above)
Comment (ishant002) [reply src/test/java/com/znsio/sample/e2e/businessLayer/amazon/AmazonHomePageBL.java:35]: @vishal07111998 the above function is created in AmazonHomePageBL class and it's return SearchResultListPageBL class object.

File: src/test/java/com/znsio/sample/e2e/screen/amazon/IphoneListScreen.java [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+                .getId());
+
+        switch(platform) {
+            case web:
+                return new IphoneListScreenWeb(driver, visually);
+        }
+        throw new NotImplementedException(SCREEN_NAME + " is not implemented in " + Runner.platform);
+    }
+
+    public abstract boolean verifyProductName();
```
[the remark this answers is not in the document]
Comment (ishant002) [reply src/test/java/com/znsio/sample/e2e/screen/amazon/IphoneListScreen.java]: @vishal07111998 I am fetching the product name and later comparing it with "iphone 13" text in the BL. So it seems to be correct I think. Please let me know if I am wrong somewhere

[the remark this answers is not in the document]
Comment (ishant002) [reply src/test/java/com/znsio/sample/e2e/screen/amazon/IphoneListScreen.java]: @vishal07111998 This function is returning String, not sure why it reflecting like this here. But in my code it's correct. No worries I'll push the updated code, then you can check.

**Pull request closed without merging**